config/globals.js

```diff
@@ -1,12 +1,127 @@
+		break;
+	case ENVIRONMENTS.TEST:
+		defaultLogLevel = 'emerg';
+		break;
+	case ENVIRONMENTS.DEVELOPMENT:
+	case ENVIRONMENTS.MIGRATION:
+	default:
+		defaultLogLevel = 'debug';
+}
+
+let defaultDbUrl = null;
+switch (NODE_ENV) {
+	case ENVIRONMENTS.TEST:
+		defaultDbUrl = 'mongodb://127.0.0.1:27017/schulcloud-test';
+		break;
+	default:
+		defaultDbUrl = 'mongodb://127.0.0.1:27017/schulcloud';
+}
+
+
+const globals = {
 	BODYPARSER_JSON_LIMIT: process.env.BODYPARSER_JSON_LIMIT || '20mb',
 	DATABASE_AUDIT: process.env.DATABASE_AUDIT || 'false',
+	DB_URL: process.env.DB_URL || defaultDbUrl,
 	DOCUMENT_BASE_DIR: process.env.DOCUMENT_BASE_DIR || 'https://s3.hidrive.strato.com/schul-cloud-hpi/',
 	MAXIMUM_ALLOWABLE_TOTAL_ATTACHMENTS_SIZE_BYTE: (5 * 1024 * 1024), // 5MB
-	REQUEST_TIMEOUT: 8000,
+	REQUEST_TIMEOUT: process.env.REQUEST_TIMEOUT || 8000,
 	METRICS_PATH: process.env.METRICS_PATH || '/metrics',
 	MONGOOSE_CONNECTION_POOL_SIZE: parseInt(process.env.MONGOOSE_CONNECTION_POOL_SIZE || '10', 10),
 	STUDENT_TEAM_CREATE_DISABLED: process.env.STUDENT_TEAM_CREATE_DISABLED,
+
+	SC_DOMAIN: process.env.SC_DOMAIN || 'localhost',
 	SC_THEME: process.env.SC_THEME || 'default',
+	SC_TITLE: process.env.SC_TITLE || 'HPI Schul-Cloud',
+	SC_SHORT_TITLE: process.env.SC_SHORT_TITLE || 'Schul-Cloud',
 	SMTP_SENDER: process.env.SMTP_SENDER || 'noreply@schul-cloud.org',
+
+	KEEP_ALIVE: process.env.KEEP_ALIVE || false,
+	/**
+	 * default value 'development' matches default of app.get('env'), but use globals
+	 */
+	NODE_ENV,
+	ENVIRONMENTS,
+	LOG_LEVEL: process.env.LOG_LEVEL || defaultLogLevel,
+	/** HOST=https://schul-cloud.org in bosscloud, see misuse in todo */
+	HOST: process.env.HOST || 'localhost:3030', // todo this is client url in config
+	TOKEN_SUB: process.env.TOKEN_SUB
+	|| process.env.HOST || 'https://schulcloud-thueringen.de', // added TOKEN_SUB on env
+	TOKEN_ISS: process.env.TOKEN_ISS || process.env.SC_DOMAIN || 'schulcloud-thueringen.de', // added TOKEN_ISS on env
+	SYSTEM_LOG_LEVEL: process.env.SYSTEM_LOG_LEVEL || 'sendRequests',
+	// secrets smtp
+	SMTP: process.env.SMTP,
+	SMTP_HOST: process.env.SMTP_HOST,
+	SMTP_PORT: process.env.SMTP_PORT,
+	// secrets aws
+	AWS_ACCESS_KEY: process.env.AWS_ACCESS_KEY,
+	AWS_SECRET_ACCESS_KEY: process.env.AWS_ACCESS_KEY,
+	AWS_REGION: process.env.AWS_REGION || 'eu-de',
+	AWS_ENDPOINT_URL: process.env.AWS_ENDPOINT_URL,
+	AUTHENTICATION: process.env.AUTHENTICATION,
+	//
+	DISPLAY_REQUEST_LEVEL: Number(process.env.DISPLAY_REQUEST_LEVEL || 0),
+	ANALYTICS_LOGGING: process.env.ANALYTICS_LOGGING,
+	LOGIN_BLOCK_TIME: process.env.LOGIN_BLOCK_TIME || 15, // allowedTimeDifference
+	CONSENT_WITHOUT_PARENTS_MIN_AGE_YEARS: parseInt(process.env.CONSENT_WITHOUT_PARENTS_MIN_AGE_YEARS || 16, 10),
+
+	/** used in tests only currently */
+	OAUTH_URL: process.env.OAUTH_URL,
+
+	// test user helper
+	TEST_PW: (process.env.TEST_PW || '').trim(),
+	TEST_HASH: (process.env.TEST_HASH || '').trim(),
+	AT: '@schul-cloud.org',
```

Comment:
Siehe unten im Test, AT gehört für mich nicht in die config.

Reply:
Könnte man als mock-email oder so verwenden aber habe es erst mal wieder herausgeworfen.